assets/js/components/ExecutionResults/CheckResultDetail/CheckResultDetailPage.test.jsx

```diff
@@ -0,0 +1,166 @@
+            id: agent1,
+            cluster_id: clusterID,
+            hostname: hostname1,
+          }),
+          hostFactory.build({
+            id: agent2,
+            cluster_id: clusterID,
+            hostname: hostname2,
+          }),
+          hostFactory.build(),
+        ],
+      },
+      catalog: {
+        loading: false,
+        data: checksCatalog,
+        error: null,
+      },
+      lastExecutions: {
+        [clusterID]: {
+          loading: false,
+          data: completedExecution,
+          error: null,
+        },
+      },
+    };
+    return initialState;
+  };
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should not render CheckResultDetailPage when clusterID in the url is false', () => {
+    const reduxStore = initialStore();
+    const falseClusterID = faker.animal.bear();
+    const [StatefulCheckResultDetailPage] = withState(
+      <CheckResultDetailPage />,
+      reduxStore
+    );
+    renderWithRouterMatch(StatefulCheckResultDetailPage, {
+      path: 'clusters/:clusterID/executions/last/:checkID/:targetType/:targetName',
+      route: `/clusters/${falseClusterID}/executions/last/${reduxStore.catalog.data[0].id}/host/${reduxStore.hostsList.hosts[0].hostname}`,
+    });
+
+    expect(screen.getByText('Go back to cluster overview')).toBeTruthy();
+    fireEvent.click(screen.getByText('Go back to cluster overview'));
+    expect(window.location.pathname).toEqual('/clusters/');
+  });
+
+  it('should not render CheckResultDetailPage when checkID in the url is false', () => {
+    const reduxStore = initialStore();
+    const falseCheckID = faker.animal.bear();
+    const [StatefulCheckResultDetailPage] = withState(
+      <CheckResultDetailPage />,
+      reduxStore
+    );
+
+    renderWithRouterMatch(StatefulCheckResultDetailPage, {
+      path: 'clusters/:clusterID/executions/last/:checkID/:targetType/:targetName',
+      route: `/clusters/${reduxStore.clustersList.clusters[0].id}/executions/last/${falseCheckID}/host/${reduxStore.hostsList.hosts[0].hostname}`,
+    });
+
+    expect(screen.getByText('Go back to last execution')).toBeTruthy();
+    fireEvent.click(screen.getByText('Go back to last execution'));
+    expect(window.location.pathname).toEqual(
+      `/clusters/${reduxStore.clustersList.clusters[0].id}/executions/last`
+    );
+  });
+  it('should not render CheckResultDetailPage when targetType in the url is false', () => {
+    const reduxStore = initialStore();
+    const invalidTargetType = 'falseTargetType';
+    const [StatefulCheckResultDetailPage] = withState(
+      <CheckResultDetailPage />,
+      reduxStore
+    );
+    renderWithRouterMatch(StatefulCheckResultDetailPage, {
+      path: 'clusters/:clusterID/executions/last/:checkID/:targetType/:targetName',
+      route: `/clusters/${reduxStore.clustersList.clusters[0].id}/executions/last/${reduxStore.catalog.data[0].id}/${invalidTargetType}/${reduxStore.hostsList.hosts[0].hostname}`,
```

Comment:
I think `reduxStore.clustersList.clusters[0].id` can be declared as a constant. Same goes for other variables

Reply:
Fair point, i wrote a helper function for that!